### PR TITLE
Improve method categorization / handler branching logic

### PR DESCRIPTION
**Pull request by bangtoven**

### CODE REVIEW

File: packages/wallet-sdk/src/core/provider/util.ts [hunk 1]
```diff
@@ -1,13 +1,34 @@
-import { standardErrors } from './error';
-import { ProviderInterface, RequestArguments } from './type/ProviderInterface';
+import { LIB_VERSION } from 'src/version';
+
+import { standardErrors } from '../error';
+import { ProviderInterface, RequestArguments } from './interface';
+import { Chain } from ':core/type';
+
+export async function fetchRPCRequest(request: RequestArguments, chain: Chain) {
```
Comment (bangtoven) [inline packages/wallet-sdk/src/core/provider/util.ts:7]: replacing RPCFetchRequestHandler https://github.com/coinbase/coinbase-wallet-sdk/pull/1203/files#diff-3b439c47c25659fd35e5255a60d451441122df80ea8c2c18e97a945befb5c4af

File: packages/wallet-sdk/src/CoinbaseWalletProvider.test.ts [hunk 2]
```diff
@@ -31,9 +31,7 @@ describe('CoinbaseWalletProvider', () => {
   describe('Request Handling', () => {
     test('handles request correctly', async () => {
       const response1 = await provider.request({ method: 'eth_chainId' });
-      const response2 = await provider.request({ method: 'eth_accounts' });
       expect(response1).toBe(1);
-      expect(response2).toEqual([]);
```
Comment (bangtoven) [inline packages/wallet-sdk/src/CoinbaseWalletProvider.test.ts:36]: this test was wrong. when it's not connected, `eth_accounts` should throw error.

File: packages/wallet-sdk/src/CoinbaseWalletProvider.ts [hunk 3]
```diff
@@ -61,24 +42,58 @@ export class CoinbaseWalletProvider extends EventEmitter implements ProviderInte
...
+    sign: (request: RequestArguments) =>
+      this.signRequestHandler.handleRequest(request, this.accounts),
+
+    filter: (request: RequestArguments) => {
+      const filterHandler = new FilterRequestHandler(this.handlers.fetch);
+      return filterHandler.handleRequest(request);
+    },
+
+    state: (request: RequestArguments) => {
+      const getConnectedAccounts = (): AddressString[] => {
```
Comment (bangtoven) [inline packages/wallet-sdk/src/CoinbaseWalletProvider.ts:69]: migrated from StateRequestHandler

File: packages/wallet-sdk/src/filter/FilterPolyfill.ts [hunk 4]
```diff
@@ -110,38 +113,6 @@ export class FilterPolyfill {
...
-          return reject(err);
-        }
-        if (Array.isArray(response) || response == null) {
-          return reject(new Error(`unexpected response received: ${JSON.stringify(response)}`));
-        }
-        resolve(response);
-      });
-    });
-  }
-
```
Comment (bangtoven) [inline packages/wallet-sdk/src/filter/FilterPolyfill.ts:144]: 🤯

File: packages/wallet-sdk/src/core/provider/method.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
+const mapping = {
```
Comment (bangtoven) [inline packages/wallet-sdk/src/core/provider/method.ts:1]: consolidating all method categorization info,
used to be `canHandle` function on each "RequestHandler"

**Review by fan-zhang-sv — APPROVED** [review]

**Pull request merged**